fido2/server.py

```diff
@@ -49,13 +49,33 @@ class ATTESTATION(six.text_type, Enum):
     DIRECT = 'direct'
 
 
+@unique
+class USER_VERIFICATION(six.text_type, Enum):
+    DISCOURAGED = 'discouraged'
+    PREFERRED = 'preferred'
+    REQUIRED = 'required'
+
+
 class Fido2Server(object):
-    def __init__(self, rp, attestation=None, verify_origin=None):
+    def __init__(
+            self,
+            rp,
+            attestation=None,
+            verify_origin=None,
+            resident_key=False,
+            user_verification=USER_VERIFICATION.PREFERRED
+    ):
+        if user_verification not in USER_VERIFICATION:
+            raise ValueError('"{}" is not a member of {}'
+                             .format(user_verification, USER_VERIFICATION))
+
         self.rp = rp
         self._verify = verify_origin or _verify_origin_for_rp(rp['id'])
         self.timeout = 30
         self.attestation = attestation or ATTESTATION.NONE
```

Comment:
Come to think of it, we should probably do `self.attestation = ATTESTATION(attestation)` here as well, and change the default from `None` to `ATTESTATION.NONE`. Would you agree?

Reply:
Agreed.